### ISSUE TITLE
Example: How to I deauthorize a token?

### ISSUE THREAD

**Issue by jeffblake** [oauth-xx/oauth2#224]
Many OAuth providers have the ability to disconnect a token. E.g.:

Quickbooks:
https://developer.intuit.com/docs/0150_payments/0060_authentication_and_authorization/oauth_management_api

Stripe: 
https://stripe.com/docs/connect/reference

Is there a roadmap for integrating this into the oauth2 gem?

**Comment by xtagon** [comment]
Is this the same as https://github.com/oauth-xx/oauth2/issues/244?

**Comment by pboling** [comment]
Yes, this is the same as #244.  This one was opened first, so closing the other.

**Comment by pboling** [comment]
Ref from the closed #244 - http://tools.ietf.org/html/rfc7009

**Comment by ionutzp** [comment]
👍

**Comment by Onumis** [comment]
Looking into this because I want to programmatically revoke depending on some conditions.
For the Google provider, for example, they inform the following:

```ruby
# To programmatically revoke a token, make an HTTP request to the oauth2.revoke endpoint:

uri = URI('https://accounts.google.com/o/oauth2/revoke')
params = { :token => auth_client.access_token }
uri.query = URI.encode_www_form(params)
response = Net::HTTP.get(uri)
```

https://developers.google.com/identity/protocols/OAuth2WebServer#tokenrevoke

---

So, I was looking into the source code and found that we can just do the following:

```
# Instantiate the token
token = OAuth2::AccessToken.new(omniauth_strategy.client,  access_token, refresh_token: refresh_token)

# revoke
token.post('/o/oauth2/revoke', params: { token: user.refresh_token })
```

I'm using the gems omniauth and omniauth-google-oauth2, but that should not matter.
Seems simple enough for anyone to do this, but let me know if it's still worth it's own method, very much like the `refresh` method.

**Comment by pboling** [comment]
Thanks for the tips!  We should get that idea into documentation if it isn't already, as it seems to be a frequent question.  Leaving this open until we get it documented (aside from here in the issue tracker).

**Comment by pboling** [comment]
I'd love to get this in the wiki, but I don't have time to myself.  Closing for now.

**Comment by pboling** [comment]
Actually, I am going to add a `revoke` method.  Tracking on GitLab here: https://gitlab.com/oauth-xx/oauth2/-/issues/224

**Comment by pboling** [comment]
Fixed by https://github.com/oauth-xx/oauth2/pull/643

**Comment by pboling** [comment]
This issue is a copy of https://gitlab.com/oauth-xx/oauth2/-/issues/224
